### PR TITLE
[vcpkg docs] Remove deprecated functions from docs

### LINKED ISSUE #17178: Don't use deprecated funtions in your examples
In the documentation always the deprecated functions `vcpkg_..._cmake` are used.
****

**Is your feature request related to a problem? Please describe.**
Often people use the deprecated functions (and the documentation suggest that)

**Proposed solution**
Use `vcpkg_cmake_...` in the documentation and add that you need a host dependency to the right packages. Maybe add a full example/step by step tutorial how to add an cmake port.

**Describe alternatives you've considered**
None

**Additional context**
Search for `_cmake(` in VS Code with _files to include_ = `*.md`

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17178
  Remove use of deprecated cmake functions in the docs
  Use quotes for paths in examples 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

### CODE REVIEW

**Review by ras0219-msft — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@Thomas1664 ,Thanks for your pr, It looks like CI is failing because of format issues, could you please take a look?
````
Starting: Create Diff
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.198.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
Formatted command: . 'C:\a\1\s\scripts\azure-pipelines\Create-PRDiff.ps1' -DiffFile 'C:\a\1\a\format.diff'
========================== Starting Command Output ===========================
"C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\a\_temp\d4f86608-513d-41dd-a143-1ed4b4d61128.ps1'"
C:\a\1\s\scripts\azure-pipelines\Create-PRDiff.ps1 : The formatting of the files in the repo were not what we expected,
or the documentation was not regenerated.
Please access the diff from format.diff in the build artifacts,
and apply the patch with `git apply`
At C:\a\_temp\d4f86608-513d-41dd-a143-1ed4b4d61128.ps1:3 char:1
+ . 'C:\a\1\s\scripts\azure-pipelines\Create-PRDiff.ps1' -DiffFile 'C:\ …
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
+ FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Create-PRDiff.ps1
##[error]PowerShell exited with code '1'.
Finishing: Create Diff

````

**Comment by Thomas1664** [comment]
@JonLiu1993 CI wants me to revert my changes:
```diff
diff --git a/docs/maintainers/vcpkg_check_features.md b/docs/maintainers/vcpkg_check_features.md
index c22908589..a8b90a3cd 100644
--- a/docs/maintainers/vcpkg_check_features.md
+++ b/docs/maintainers/vcpkg_check_features.md
@@ -56,8 +56,9 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         secure    MI_SECURE
 )
 
-vcpkg_cmake_configure(
+vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
     OPTIONS
         # Expands to "-DMI_SEE_ASM=ON;-DMI_OVERRIDE=OFF;-DMI_SECURE=ON"
         ${FEATURE_OPTIONS}
@@ -76,8 +77,9 @@ vcpkg_check_features(
         websockets  CPPREST_EXCLUDE_WEBSOCKETS
 )
 
-vcpkg_cmake_configure(
+vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
     OPTIONS
         # Expands to "-DCPPREST_EXCLUDE_BROTLI=ON;-DCPPREST_EXCLUDE_WEBSOCKETS=OFF"
         ${FEATURE_OPTIONS}
@@ -97,8 +99,9 @@ vcpkg_check_features(
         cuda  BUILD_GPU
 )
 
-vcpkg_cmake_configure(
+vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
     OPTIONS
         # Expands to "-DWITH_CUDA=ON;-DBUILD_CUDA=ON;-DBUILD_GPU=ON"
         ${FEATURE_OPTIONS}
@@ -118,8 +121,9 @@ vcpkg_check_features(
         tbb   ROCKSDB_IGNORE_PACKAGE_TBB
 )
 
-vcpkg_cmake_configure(
+vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
     OPTIONS
         # Expands to "-DWITH_TBB=ON;-DROCKSDB_IGNORE_PACKAGE_TBB=OFF"
         ${FEATURE_OPTIONS}
diff --git a/docs/maintainers/vcpkg_find_fortran.md b/docs/maintainers/vcpkg_find_fortran.md
index b02c0c51b..227c1c837 100644
--- a/docs/maintainers/vcpkg_find_fortran.md
+++ b/docs/maintainers/vcpkg_find_fortran.md
@@ -15,7 +15,7 @@ vcpkg_find_fortran(<out_var>)
 ```cmake
 vcpkg_find_fortran(fortran_args)
 # ...
-vcpkg_cmake_configure(...
+vcpkg_configure_cmake(...
     OPTIONS
         ${fortran_args}
 )

```

**Comment by Thomas1664** [comment]
@JonLiu1993 
I found out that I needed to edit the .cmake files and not the .md files. Do I really need to revert the changes in those .md files?

**Comment by dg0yt** [comment]
Nothing needed to revert. Just apply the original changes to the scripts. Then regenerate the doc files with the powershell script in the doc dir.

**Comment by Thomas1664** [comment]
> Nothing needed to revert. Just apply the original changes to the scripts. Then regenerate the doc files with the powershell script in the doc dir.

I applied the changes in the scripts. The powershell script doesn't work because I get a bunch of errors.

**Comment by Thomas1664** [comment]
> That's a pity. It works in CI and even on Linux.

Hopefully I don't need to run it because I edited the docs already directly. Lets see if CI complains...

**Comment by Thomas1664** [comment]
Great! Now that I edited the scripts CI builds **every** port

**Comment by Thomas1664** [comment]
> Thanks for your pr, It looks like CI is failing because of format issues, could you please take a look?

@JonLiu1993 , `x86-windows` succeeded, no need to run the ps script.

**Review by JackBoosY — APPROVED** [review]

File: docs/examples/packaging-github-repos.md [hunk 1]
```diff
@@ -32,15 +32,12 @@ The important parts to update are `REPO` for the GitHub repository path, `REF` f
...
-)
-vcpkg_install_cmake()
-file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/libogg RENAME copyright)
+vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH})
+vcpkg_cmake_install()
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/libogg" RENAME copyright)
 ```
 
-Check the documentation for [`vcpkg_configure_cmake`](../maintainers/vcpkg_configure_cmake.md) and [`vcpkg_install_cmake`](../maintainers/vcpkg_install_cmake.md) if your package needs additional options. 
+Check the documentation for [`vcpkg_cmake_configure`](../maintainers/vcpkg_cmake_configure.md) and [`vcpkg_cmake_install`](../maintainers/vcpkg_cmake_install.md) if your package needs additional options.
```
Comment (BillyONeal) [inline docs/examples/packaging-github-repos.md:40]: This link is broken; there are probably others. Can you check that they are connected?

**Review by BillyONeal — CHANGES_REQUESTED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Cycled the PR in order to get CLA bot to be happy

**Pull request merged**